doctr/models/recognition/crnn/pytorch.py

```diff
@@ -0,0 +1,225 @@
+            # collapse best path (using itertools.groupby), map to chars, join char list to string
+            collapsed = [vocab[k] for k, _ in groupby(sequence) if k != blank]
+            res = ''.join(collapsed)
+            words.append(res)
+
+        return list(zip(words, probs.tolist()))
+
+    def __call__(
+        self,
+        logits: torch.Tensor
+    ) -> List[Tuple[str, float]]:
+        """
+        Performs decoding of raw output with CTC and decoding of CTC predictions
+        with label_to_idx mapping dictionnary
+
+        Args:
+            logits: raw output of the model, shape BATCH_SIZE X NUM_CLASSES + 1 X SEQ_LEN
+
+        Returns:
+            A tuple of 2 lists: a list of str (words) and a list of float (probs)
+
+        """
+        # Decode CTC
+        return self.ctc_best_path(logits=logits, vocab=self.vocab, blank=len(self.vocab))
+
+
+class CRNN(RecognitionModel, nn.Module):
+    """Implements a CRNN architecture as described in `"An End-to-End Trainable Neural Network for Image-based
+    Sequence Recognition and Its Application to Scene Text Recognition" <https://arxiv.org/pdf/1507.05717.pdf>`_.
+
+    Args:
+        feature_extractor: the backbone serving as feature extractor
+        vocab: vocabulary used for encoding
+        rnn_units: number of units in the LSTM layers
+        cfg: configuration dictionary
+    """
+
+    _children_names: List[str] = ['feat_extractor', 'decoder', 'postprocessor']
+
+    def __init__(
+        self,
+        feature_extractor: nn.Module,
+        vocab: str,
+        rnn_units: int = 128,
+        cfg: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__(vocab=vocab, cfg=cfg, max_length=32)
+        self.feat_extractor = feature_extractor
+
+        self.decoder = nn.LSTM(
+            input_size=1 * 512, hidden_size=rnn_units, batch_first=True, num_layers=2, bidirectional=True
+        )
+
+        # features units = 2 * rnn_units because bidirectional layers
+        self.linear = nn.Linear(in_features=2 * rnn_units, out_features=len(vocab) + 1)
+
+        self.postprocessor = CTCPostProcessor(vocab=vocab)
+
+    def compute_loss(
+        self,
+        model_output: torch.Tensor,
+        target: List[str],
+    ) -> torch.Tensor:
+        """Compute CTC loss for the model.
+
+        Args:
+            gt: the encoded tensor with gt labels
+            model_output: predicted logits of the model
+            seq_len: lengths of each gt word inside the batch
+
+        Returns:
+            The loss of the model on the batch
+        """
+        gt, seq_len = self.compute_target(target)
+        batch_len = model_output.shape[0]
+        input_length = model_output.shape[1] * torch.ones(size=(batch_len,), dtype=torch.int32)
+        # N x T x C -> T x N x C
+        logits = model_output.permute(1, 0, 2)
+        probs = F.log_softmax(logits, dim=-1)
+        ctc_loss_fn = nn.CTCLoss(blank=len(self.vocab))
+        ctc_loss = ctc_loss_fn(
```

Comment:
```python
ctc_loss = F.ctc_loss(
    probs, torch.from_numpy(gt), input_length, torch.tensor(seq_len, dtype=torch.int), len(self.vocab)
)
```
if you want to do everything in the call

otherwise instantiate the ctcloss module in the constructor ;)